sdk/python/starwhale/wrapper.py

```diff
@@ -0,0 +1,94 @@
+from typing import Iterator, Any, Optional
+
+from . import data_store
+
+import os
+import re
+import threading
+
+
+class Logger:
+    def _init_writers(self, tables: list[str]):
+        self._writers = {table: None for table in tables}
+        self._lock = threading.Lock()
+
+    def close(self) -> None:
+        with self._lock:
+            for writer in self._writers.values():
+                if writer is not None:
+                    writer.close()
+
+    def _log(self, table_name: str, record: dict[str, Any]) -> None:
+        with self._lock:
+            writer = self._writers[table_name]
+            if writer is None:
+                writer = data_store.TableWriter(table_name)
+                self._writers[table_name] = writer
+        writer.insert(record)
+
+
+class Evaluation(Logger):
+    def __init__(self, eval_id: Optional[str] = None):
+        if eval_id is None:
+            eval_id = os.getenv("SW_EVAL_ID", None)
+        if eval_id is None:
+            raise RuntimeError("eval id should not be None")
+        if re.match(r"^[A-Za-z0-9-_]+$", eval_id) is None:
+            raise RuntimeError(
+                f"invalid eval id {eval_id}, only letters(A-Z, a-z), digits(0-9), hyphen('-'), and underscore('_') are allowed"
+            )
+        self.eval_id = eval_id
+        self.project = os.getenv("SW_PROJECT")
+        if self.project is None:
+            raise RuntimeError("SW_PROJECT is not set")
+        self._results_table_name = f"project/{self.project}/eval/{self.eval_id}/results"
+        self._summary_table_name = f"project/{self.project}/eval/summary"
+        self._init_writers([self._results_table_name, self._summary_table_name])
+        self._data_store = data_store.get_data_store()
+
+    def log_result(self, data_id: str, result: Any, **kwargs) -> None:
+        record = {"id": data_id, "result": result}
+        for k, v in kwargs.items():
+            record[k.lower()] = v
+        self._log(self._results_table_name, record)
+
+    def log_metrics(self, metrics: Optional[dict[str, Any]] = None, **kwargs) -> None:
+        record = {"id": self.eval_id}
+        if metrics is not None:
+            for k, v in metrics.items():
+                k = k.lower()
+                if k != "id":
+                    record[k] = v
+        else:
+            for k, v in kwargs.items():
+                record[k.lower()] = v
+        self._log(self._summary_table_name, record)
+
+    def get_results(self) -> Iterator[dict[str, Any]]:
+        return self._data_store.scan_tables([(self._results_table_name, "result", False)])
+
+
+class Dataset(Logger):
```

Comment:
shall we introduce another lower level dataset which is independent of project. Mybe the one here could be called `SWDataset`, while that lower level one could be called `Dataset`

Reply:
I prefer Dataset because it is recommended to use it like this:
```
import starwhale as sw
ds = sw.Dataset()
```
`sw.SWDataset` looks really weird

Reply:
I didn't take account the use case. You are right!